Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -1991,7 +1991,17 @@ extension URL {
 
 #if !NO_FILESYSTEM
     private static func isDirectory(_ path: String) -> Bool {
-        return FileManager.default._fileStat(path)?.isDirectory ?? false
+#if !FOUNDATION_FRAMEWORK
+        var isDirectory: Bool = false
+#else
+        var isDirectory: ObjCBool = false
+#endif
+        _ = FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
```

Comment:
I think we could maybe just group it for clarity
```
#if FOUNDATION_FRAMEWORK
var isDirectory: ObjCBool = false
_ = FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
return isDirectory.boolValue
#else
...
#endif
```
but that's really a minor nit. Thanks for fixing this!